src/snowflake/snowpark/modin/plugin/utils/numpy_to_pandas.py

```diff
@@ -112,6 +112,36 @@ def may_share_memory_mapper(a: Any, b: Any, max_work: Optional[int] = None) -> b
     return False
 
 
+def full_like_mapper(
+    a: Union[pd.DataFrame, pd.Series],
+    fill_value: Hashable,
+    dtype: Optional[Any] = None,
+    order: Optional[str] = "K",
+    subok: Optional[bool] = True,
+    shape: Optional[tuple[Any]] = None,
+) -> Union[pd.DataFrame, pd.Series]:
+    if not subok:
+        return NotImplemented
+    if not order == "K":
+        return NotImplemented
+    if dtype is not None:
+        return NotImplemented
+
+    result_shape = shape
+    if isinstance(result_shape, tuple) and len(result_shape) == 0:
+        result_shape = (1,)
+    if isinstance(result_shape, int):
+        result_shape = (result_shape,)
+    if result_shape is None:
+        result_shape = a.shape
+    if len(result_shape) == 2:
+        height, width = result_shape  # type: ignore
+        return pd.DataFrame(fill_value, index=range(height), columns=range(width))
```

Comment:
Don't we need to set dtype here? Also if a DataFrame contains multiple dtypes, are the returned object's columns supposed to be the same type as each of the columns mapped positionally?

Reply:
No. Everything ends up being the fill_value, and where there are mixed types the resulting dataframe is an object. I think that's a bug in pandas/numpy we don't necessarily want to keep. There's other bugs in the pandas implementation. You can use full_like with a mixed type dataframe with any full_value; but if you use it on a dataframe with only a single type the fill_value has to be the same.

The pandas implementation doesn't make a lot of sense. Here we basically just pass in the value and do the default thing.

Reply:
That makes sense to me, but can we document these divergences in behavior?